internal/api/api_test.go

```diff
@@ -202,7 +224,8 @@ func (fake *fakeGCS) RoundTrip(req *http.Request) (*http.Response, error) {
 
 	content, err := os.Open(filename)
 	if err != nil {
-		fake.Fatalf("Failed to open test data: %v", err)
+		fake.Logf("Failed to open test data: %v", err)
+		return nil, newStorageError("opening test data", err)
```

Comment:
This should probably return a 404 response, not an error (but include the error text in the response)

And since you are now returning an error/response you don't need the logf